pkg/internal/apis/certmanager/validation/certificaterequest.go

```diff
@@ -135,7 +135,7 @@ func patchDuplicateKeyUsage(usages []cmapi.KeyUsage) []cmapi.KeyUsage {
 			newUsages = append(newUsages, cmapi.UsageDigitalSignature)
 			// prevent having 2 UsageDigitalSignature in the slice
 			hasUsageSigning = true
-		} else {
+		} else if usage != cmapi.UsageSigning && usage != cmapi.UsageDigitalSignature {
```

Comment:
nit: Did you consider moving the `hasUsageSigning` check into a nested `if statement`? Then this `else if` would not be necessary.

Reply:
yes, i don't think nested if's look good in this case